controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -319,15 +319,15 @@ func (r *StorageClassClaimReconciler) reconcileConsumerPhases() (reconcile.Resul
 		// Wait until all PVs using the StorageClass under deletion are removed.
 		// Check for any PVs using the StorageClass.
 		pvList := corev1.PersistentVolumeList{}
-		err := r.list(&pvList, client.MatchingFields{
-			"spec.storageClassName": r.storageClassClaim.Name,
-		})
+		err := r.list(&pvList)
 		if err != nil {
 			return reconcile.Result{}, fmt.Errorf("failed to list PersistentVolumes: %s", err)
 		}
-		if len(pvList.Items) > 0 {
-			return reconcile.Result{}, fmt.Errorf("StorageClass %s is still in use by one or more PV(s)",
-				r.storageClassClaim.Name)
+		for _, pv := range pvList.Items {
+			if pv.Spec.StorageClassName == r.storageClassClaim.Name {
+				return reconcile.Result{}, fmt.Errorf("StorageClass %s is still in use by one or more PV(s)",
+					r.storageClassClaim.Name)
+			}
```

Comment:
This for-range loop will create an in memory copy of the PV objects, please use index instead:
```suggestion
		for i := range pvList.Items {
			pv := &pvList.Items[i]
			if pv.Spec.StorageClassName == r.storageClassClaim.Name {
				return reconcile.Result{}, fmt.Errorf("StorageClass %s is still in use by one or more PV(s)",
					r.storageClassClaim.Name)
			}
```